clients/venice-client/src/main/java/com/linkedin/venice/fastclient/DispatchingAvroGenericStoreClient.java

```diff
@@ -572,6 +571,9 @@ private void batchGetTransportRequestCompletionHandler(
     VeniceCompressor compressor =
         metadata.getCompressor(transportClientResponse.getCompressionStrategy(), requestContext.currentVersion);
     for (MultiGetResponseRecordV1 r: records) {
+      // Since the underlying FastSerializerDeserializerFactory is caching deserializers by reader, writer and object
+      // class. Retrieving the corresponding deserializer per record should be fast here after initial warmup.
+      RecordDeserializer<V> dataRecordDeserializer = getDataRecordDeserializer(r.getSchemaId());
```

Comment:
Looks like there are multiple hops inside `getDataRecordDeserializer`. Should we cache the lookup locally ?

Reply:
In other code paths we have optimized this access pattern with `AvroStoreDeserializerCache`. It is in `venice-common`, so FC should have access to it already.